pkg/apis/keepalive/v1beta1/register.go

```diff
@@ -0,0 +1,74 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// VirtualIP is a keepalived CRD specificiation for virtual IP addresses
+type VirtualIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   VirtualIPSpec `json:"spec"`
+	Status Status        `json:"status"`
+}
+
+// VirtualIPSpec defines the spec of the CRD
+type VirtualIPSpec struct {
+	// IP virtual IP to use
+	IP string `json:"ip"`
+	// Port of the virtual
+	Port int `json:"port"`
+
+	// Interface interface where the virtual IP should be configured
+	// The default value is eth0
+	Interface *string `json:"interface,omitempty"`
+
+	// ProxyProtocol indicates if proxy-protocol is enabled in the service being exposed
+	ProxyProtocol bool `json:"proxyProtocol"`
+	//Protocol defines the protocol of the service. Valid values are TCP and UDP
+	Protocol corev1.Protocol `json:"protocol"`
```

Comment:
> protocol of the service

Shouldn't this be in `ServiceReference` then?

Reply:
done